### PR TITLE
Allow to set a custom device in ASE calculator

### PR DESCRIPTION
**Pull request by Luthaf**
One more step for #430

The idea is to allow the user to specify a custom device to run the model with. This should enable testing of models on GPU (even if the rest of ASE runs on CPU). If the user does not specify a device, we try the values in `supported_devices` in order.

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [x] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://metatensor--539.org.readthedocs.build/en/539/

### CODE REVIEW

File: python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py [hunk 1]
```diff
@@ -46,53 +60,94 @@ class MetatensorCalculator(ase.calculators.calculator.Calculator):
...
+        device=None,
     ):
         """
         :param model: model to use for the calculation. This can be a file path, a
             Python instance of :py:class:`MetatensorAtomisticModel`, or the output of
             :py:func:`torch.jit.script` on :py:class:`MetatensorAtomisticModel`.
         :param check_consistency: should we check the model for consistency when
             running, defaults to False.
+        :param dtype: data type to use for the calculations. This can be set to use
+            either 32-bits or 64bits floating point values
```
Comment (sirmarcel) [inline python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py:75]: add dash after `64` ;)

File: python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py [hunk 2]
```diff
@@ -46,53 +60,94 @@ class MetatensorCalculator(ase.calculators.calculator.Calculator):
...
+        :param device: torch device to use for the calculation. If `None`, we will try
```
Comment (sirmarcel) [inline python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py:76]: Shouldn't `None` be in double slashes?

File: python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py [hunk 3]
```diff
@@ -140,12 +200,15 @@ def run_model(
...
-        types, positions, cell = _ase_to_torch_data(atoms)
+        types, positions, cell = _ase_to_torch_data(
+            atoms=atoms, dtype=self.dtype, device=self.device
+        )
         system = System(types, positions, cell)
 
         # Compute the neighbors lists requested by the model using ASE NL
         for options in self._model.requested_neighbors_lists():
             neighbors = _compute_ase_neighbors(atoms, options)
+            neighbors = neighbors.to(dtype=self.dtype, device=self.device)
```
Comment (sirmarcel) [inline python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py:211]: Isn't `neighbors` an object with mixed ints and floats? What happens here, do the ints get turned into floats?

Comment (Luthaf) [reply python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py:211]: Only the values (here the distance vectors) of the `TensorBlock` are moved to another dtype. The Labels are always kept as `torch.int32` (but can be copied to another device).

**Comment by Luthaf** [comment]
This will be updated to only a custom device once #551 is merged

File: python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py (hunk 1 above)
Comment (frostedoyster) [inline python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py:75]: 64-bit

**Review by frostedoyster — APPROVED** [review]

**Comment by sirmarcel** [comment]
looks good to me ✅

**Pull request merged**